### ISSUE TITLE
Alternate approach to tikz  -- tikzjax

### ISSUE THREAD

**Issue by mgage** [openwebwork/pg#477]
an alternate approach to including tikz which should perhaps be looked in to.
Originally suggested by @taniwallach in issue #418.

#292
http://webwork.maa.org/moodle/mod/forum/discuss.php?d=4625
Alternate approach - tikzjax - see:

http://tikzjax.com/
http://webwork.maa.org/moodle/mod/forum/discuss.php?d=4616

**Comment by drgrice1** [comment]
For the record, I think Alex Jordan originally suggested this approach in the WeBWorK forums, and then @taniwallach referenced it here.

In any case, I also think this approach should be investigated further.  One problem with the current tikz implementation is that it does increase server load quite a bit.  The server has to run pdflatex on the code in the background.  If multiple students are accessing a problem that uses the macro and the problem has several images in it, the server load could certainly ramp up quite heavily.

The tikzjax approach offloads this work onto the clients browser.

**Comment by taniwallach** [comment]
> For the record, I think Alex Jordan originally suggested this approach in the WeBWorK forums, and then @taniwallach referenced it here.

Yes - I only created the issue on GitHub to keep this on the radar, so we could consider the alternatives.
 
I certainly agree with @drgrice1 that if the processing can be done on the client side, that would probably be very advantageous.

**Comment by drgrice1** [comment]
I have done a considerable amount of work on this.  I will put in a pull request at some point, but it is not ready yet.  See https://github.com/drgrice1/pg/tree/pg-tikzjax and https://github.com/drgrice1/webwork2/tree/pg-tikzjax.

Also see https://github.com/drgrice1/web2js/tree/ww-modifications, https://github.com/drgrice1/dvi2html/tree/ww-modifications, and https://github.com/drgrice1/tikzjax/tree/ww-modifications.

It isn't quite ready yet.  It is working well for Firefox or Chrome, but there seem to be some issues with Safari (and perhaps other browsers).  I don't have Safari to work with readily though, so it has been difficult to figure out.
